addons/website_livechat/models/website.py

```diff
@@ -16,5 +16,35 @@ def get_livechat_channel_info(self):
         """
         self.ensure_one()
         if self.channel_id:
-            return self.channel_id.sudo().get_livechat_info()
+            livechat_info = self.channel_id.sudo().get_livechat_info()
+            if livechat_info['available']:
+                livechat_request_info = self._get_livechat_request_info(self.channel_id.id)
+                if livechat_request_info:
+                    livechat_info['options'].update(livechat_request_info)
+            return livechat_info
         return {}
+
+    def _get_livechat_request_info(self, channel_id):
+        chat_request_info = {}
+        visitor_id = self.env['website.visitor']._get_visitor_from_request().id
```

Comment:
I see a use case where the `_get_visitor_from_request` returns `None`.
Wouldn't is be safer to store the visitor, check with `if visitor`, and then do `visitor.id` in your domain?